### PR TITLE
Use per-test individual registry key so that tests can be run in parallel

### PR DESCRIPTION
**Pull request by larskanis**
Regarding conversation in #5:

> >        I'm not sure what's better to use volatile environment and (unused_value)[https://github.com/ruby/ruby/blob/master/test/win32/test_registry.rb#L86] block. This repository don't use that. What's do you think that?

> Using a random subkey versus a fixed one is probably necessary to run the tests in parallel. This is no problem in this repository, but in ruby/ruby. I can change this.

### CODE REVIEW

File: test/win32/test_registry.rb [hunk 1]
```diff
@@ -39,9 +52,9 @@ def test_predefined
     end
 
     def test_open_no_block
-      Win32::Registry::HKEY_CURRENT_USER.create(backslachs(TEST_REGISTRY_KEY)).close
+      Win32::Registry::HKEY_CURRENT_USER.create(@test_registry_key).close
 
-      reg = Win32::Registry::HKEY_CURRENT_USER.open(backslachs(TEST_REGISTRY_KEY), Win32::Registry::KEY_ALL_ACCESS)
+      reg = Win32::Registry::HKEY_CURRENT_USER.open(@test_registry_key, Win32::Registry::KEY_ALL_ACCESS)
```
Comment (nobu) [inline test/win32/test_registry.rb:57]: Since registry keys can include `/`, it is wrong to use `File` methods to split them.

**Review by nobu — CHANGES_REQUESTED** [review]

File: test/win32/test_registry.rb (hunk 1 above)
Comment (larskanis) [reply test/win32/test_registry.rb:57]: It's only used for testing, where we have full control about our test keys. Nevertheless I'll change it.

Comment (nobu) [reply test/win32/test_registry.rb:57]: My concern is that it might encourage to write wrong code in other places.

**Comment by nobu** [comment]
I thought it would be safer to use volatile environment registries, which disappear at reboot, to test.
Not only teardown, I think registry database file bloat is a well-known problem.
Is there anything that can't be tested on volatile environment registries?

**Comment by hsbt** [comment]
FYI: I merged `win32-registry` to `ruby/ruby` with the current master HEAD revision at https://github.com/ruby/ruby/pull/11791.

I prefer [this approach](https://github.com/ruby/ruby/commit/9f8e87c44bc87f5b37364707d1f570da1d7a6f89#diff-aff06b23cb6641df56f2089a7f100b01526904c02e2112e7de0723bf345e2e03R86).

**Comment by larskanis** [comment]
I opened a new PR in #8 using volatile environment and avoiding `File` methods.

**Comment by nobu** [comment]
Close in favor of #8.

**Pull request closed without merging**